src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,845 @@
+        
+    def _isMoving(self):
+        return time.time() < self._current_move_finish
+
+    def _doMove(self, axis, new_pos):
+        # Check that the position is within the range.
+        if new_pos >= self._range[0] and new_pos <= self._range[1]:
+            self._target_pos[axis] = new_pos
+            self._start_pos = copy.copy(self._pos)
+            d = self._target_pos[axis] - self._start_pos[axis]
+            dur = driver.estimateMoveDuration(abs(d), self._speed[axis], self._accel[axis])
+            self._current_mvoe_start = time.time()
+            self._current_move_finish = time.time() + dur
+        else:
+            if new_pos > self._range[1]:
+                self._addError(axis * 100 + 4)  # Error - detected positive limit
+                self._pos[axis] = self._range[1]
+            elif new_pos < self._range[0]:
+                self._addError(axis * 100 + 5)  # error - detected negative limit
+                self._pos[axis] = self._range[0]
+
+    def _parseMessage(self, msg):
+        """
+        msg (str): the message to parse (without the \r)
+        return None: self._output_buf is updated if necessary
+        """
+        logging.debug("SIM: parsing %s", msg)
+        msg = msg.strip()  # remove leading and trailing whitespace
+        msg = "".join(msg.split())  # remove all space characters
+        
+        if msg == "VE?":
+            self._sendAnswer("1.0")
+            
+        elif msg == "SM":
+            # save memory to non-volatile RAM
+            pass
+
+        elif msg == "TE?":
+            # Query error code. Return no error code
+            if len(self._error_stack) > 0:
+                self._sendAnswer(self._error_stack.pop())  # pop the top element of the error stack
+            else:
+                self._sendAnswer(0)  # no error
+
+        # Query the axis ID number
+        elif re.match('\dID\?', msg):
+            self._sendAnswer("12345")
+
+        # Query absolute position
+        elif re.match('\dTP\?', msg):
+            axis = int(msg[0])
+            self._updateCurrentPosition()
+            self._sendAnswer(str(self._pos[axis]))
+
+        # Query current target
+        elif re.match('\dDP\?', msg):
+            axis = int(msg[0])
+            self._sendAnswer(str(self._target_pos[axis]))
+
+        # Query current speed
+        elif re.match('\dVA\?', msg):
+            axis = int(msg[0])
+            self._sendAnswer(str(self._speed[axis]))
+
+        # Query current accel
+        elif re.match('\dAC\?', msg):
+            axis = int(msg[0])
+            self._sendAnswer(str(self._accel[axis]))
+
+        # Query current decel
+        elif re.match('\dAG\?', msg):
+            axis = int(msg[0])
+            self._sendAnswer(str(self._decel[axis]))
+
+        # Query motion done
+        elif re.match('\dMD\?', msg):
+            self._updateCurrentPosition()
+
+            if time.time() > self._current_move_finish:
```

Comment:
You should be able to move this directly into the block below. However, it's probably not needed at all. What does it help to set `current_move_finish` to 0?